### DISCUSSION TITLE
Develop command doesn't work

### DISCUSSION THREAD

**Discussion by lomby92** [gatsbyjs/gatsby#30124] [Help] [answered]
## Description
After updating to the latest version of Gatsby i'm not able to run the development server with the hot-reload.
The error:
```
The 'compilation' argument must be an instance of Compilation
  TypeError: The 'compilation' argument must be an instance of Compilation
```
This seems to be an error from two webpack 5.x packages into my project, one from Gatsby and one from NetlifyCMS.

NOTE: the `gatsby build` command works and i'm able to run `gatsby serve` to access the website, that has no errors.

### Steps to reproduce
Run `gatsby develop` in a project with 2 dependencies that depend on webpack 5

### Expected result
We should be able to run `gatsby develop` with the hot reload.

### Actual result
An error occurs

### Environment
```
  System:
    OS: Linux 5.4 KDE neon 5.21
    CPU: (12) x64 Intel(R) Core(TM) i7-8750H CPU @ 2.20GHz
    Shell: 5.8 - /usr/bin/zsh
  Binaries:
    Node: 12.20.0 - ~/.nvm/versions/node/v12.20.0/bin/node
    Yarn: 1.22.5 - /usr/bin/yarn
    npm: 6.14.8 - ~/.nvm/versions/node/v12.20.0/bin/npm
  Languages:
    Python: 2.7.18 - /usr/bin/python
  Browsers:
    Chrome: 89.0.4389.82
    Firefox: 86.0
  npmPackages:
    gatsby: ^3.0.3 => 3.0.3
    gatsby-image: ^2.11.0 => 2.11.0
    gatsby-plugin-manifest: ^3.0.0 => 3.0.0
    gatsby-plugin-netlify-cms: ^5.0.0 => 5.0.0
    gatsby-plugin-postcss: ^4.0.0 => 4.0.0
    gatsby-plugin-react-helmet: ^4.0.0 => 4.0.0
    gatsby-plugin-sharp: ^3.0.0 => 3.0.0
    gatsby-plugin-sitemap: ^3.0.0 => 3.0.0
    gatsby-remark-images: ^4.0.0 => 4.0.0
    gatsby-remark-relative-images: ^2.0.2 => 2.0.2
    gatsby-source-filesystem: ^3.0.0 => 3.0.0
    gatsby-transformer-remark: ^3.0.0 => 3.0.0
    gatsby-transformer-sharp: ^3.0.0 => 3.0.0
  npmGlobalPackages:
    gatsby-cli: 3.0.0
```

**Comment by LekoArts** [comment, marked as answer]
Hi!

I've set up a minimal project with Netlify CMS and it works. I think you're running into the same thing as https://github.com/gatsbyjs/gatsby/issues/29975

If you run `yarn why webpack` you'll probably have multiple versions in your project. Please see https://www.gatsbyjs.com/docs/reference/release-notes/migrating-from-v2-to-v3/#handling-version-mismatches on how to resolve it.

**Reply by lomby92** [reply]
Thanks, i have run:
- `rm -rf node_modules`
- `rm package-lock.json` This is important, the fix doesn't work without this
- `npm install` as i have npm 6.x

Now i'm able to run `develop` without the error

**Reply by LekoArts** [reply]
Great!

**Reply by ownchoice** [reply]
Oh, wow, I completely forgot to delete the `package-lock.json` file while cleaning after a failed update. Now it works.

**Reply by louishugens** [reply]
If you are using yarn, you should remove yarn.lock.

**Reply by polnikale** [reply]
Are you sure it should be chosen as an answer? Deleting lock file is a destructive operation as it may result in bumping a lot of other libraries. There should be a more elegant solution

**Reply by davechurilla** [reply]
Doesn't work. Still get the error.

**Reply by doublejosh** [reply]
I get this error on Gatsby Cloud, but it works fine locally (and on Netlify) when I `npm run build`.
